package.json

```diff
@@ -65,71 +65,76 @@
-    "pre_build": "echo 'Compiling typescript files'",
-    "_build": "build() { npm run compile-$1 && echo && npm run -- make-$1-components $2; }; build",
-    "_build_": "b() { echo '============='; echo Building $1:; echo '============='; npm run -- _build $1 $2; }; b",
-    "build-cjs": "npm run -s -- _build_ cjs --build",
-    "build-mjs": "npm run -s -- _build_ mjs",
-    "build": "npm run -s build-mjs",
-    "build-all": "npm run -s build-mjs ; echo ; npm run -s build-cjs",
+    "copy:mj2": "copy() { pnpm -s log:single 'Copying legacy code AsciiMath'; pnpm copyfiles -u 1 'ts/input/asciimath/legacy/**/*' $1; }; copy",
+    "copy:mml3": "copy() { pnpm -s log:single 'Copying legacy code MathML3'; pnpm copyfiles -u 1 ts/input/mathml/mml3/mml3.sef.json $1; }; copy",
+    "copy:assets": "pnpm -s log:comp 'Copying assets'; copy() { pnpm -s copy:mj2 $1 && pnpm -s copy:mml3 $1; }; copy",
+    "copy:pkg": "copy() { pnpm -s log:single \"Copying package.json to $1\"; pnpm copyfiles -u 2 components/bin/package.json $1; }; copy",
+    "=============================================================================== log": "",
+    "log:single": "log() { echo \\\\033[34m--$1\\\\033[0m; }; log",
+    "log:header": "log() { echo '============='; echo $1; echo '============='; }; log",
+    "log:comp": "log() { echo \\\\033[32m$1\\\\033[0m; }; log",
+    "=============================================================================== cjs": "",
+    "cjs:components:copy": "pnpm copyfiles -u 2 -e 'components/mjs/**/*.js' 'components/mjs/**/*' components/cjs",
+    "cjs:bundle:clean": "pnpm clean:dir bundle-cjs",
+    "cjs:components:clean": "pnpm -s log:single \"Cleaning cjs components\" pnpm rimraf components/cjs",
+    "cjs:components:compile": "pnpm -s log:single 'Compiling component files'; pnpm tsc --project tsconfig/components.json",
+    "cjs:components:finalize": "pnpm -s log:comp 'Finalize cjs components'; pnpm -s cjs:components:copy && pnpm -s copy:pkg components/cjs && pnpm -s clean:lib cjs",
+    "cjs:compile": "pnpm -s log:single 'Compiling cjs typescript files' && pnpm tsc --project tsconfig/cjs.json",
+    "cjs:bundle:finalize": "pnpm -s log:single 'Finalize cjs bundle'; echo '{\n  \"type\": \"commonjs\"\n}' > bundle-cjs/package.json;",
+    "cjs:components:make": "make() { pnpm -s log:single 'Making cjs components'; components/bin/makeAll --cjs --terse --bundle-cjs $1 components/cjs; }; make",
+    "cjs:components:src:build": "pnpm -s log:comp 'Building cjs components sources'; pnpm cjs:components:clean && pnpm cjs:components:compile && pnpm cjs:components:finalize",
+    "cjs:components:build": "pnpm -s log:comp 'Building cjs components'; pnpm -s cjs:components:src:build && pnpm clean:dir bundle-cjs && pnpm -s cjs:components:make --build",
+    "cjs:src:build": "pnpm -s log:comp 'Building cjs sources'; pnpm -s link:full && pnpm clean:dir cjs && pnpm -s cjs:compile && pnpm -s copy:assets cjs && pnpm -s copy:pkg cjs",
```

Comment:
Now I'm wondering if these lines want to be sorted so that all the `cjs.components` are together.  E.g., perhaps move `cjs:bundle:clean` and `cjs:bundle:finalize`to the top followed by `cjs:compile`, and put the other ones in an order that makes sense?  I haven't looked at what order they get used, so if that is the order they are currently in, that's OK, too.

Reply:
I've sorted all `:` scripts alphabetically.